matRad_calcDoseDirectMC.m

```diff
@@ -92,7 +92,12 @@
         % already performed in dij construction
         if size(dij.physicalDose{1},2) ~= pln.propStf.numOfBeams
             matRad_cfg.dispWarning('Number of beams stored not the same as size of dij. Using singular weight for MC');
-            dij.numOfBeams = size(dij.physicalDose{1},2);
+            % This parameter should be 1 since calcDoseDirect is true and all weights are used in the simulation already
+            if size(dij.physicalDose{1},2) ~= 1
+                matRad_cfg.dispError('Singular weight was called but dij has %i columns!',1)
+            end
+            dij.numOfBeams = 1;
+            dij.beamNum = 1;
```

Comment:
Hi @HomolkaN 
I don't think this override should be done here alone. The parameters need to be set in the respective functions to create a valid dij with one line. I will prepare a fix for this quickly. You then need to test if it works with TOPAS as well.

Reply:
both `matRad_calcParticleDoseMCsquare `and `matRad_calcParticleDoseMCtopas` generate valid dijs in both calcDoseDirect and dij mode. So this override is not needed anymore.